### DISCUSSION TITLE
Discussion on session storage and registering a new user

### DISCUSSION THREAD

**Discussion by Loganwebb1994** [nss-day-cohort-46/How-to-Ask-for-Help#23] [Ask For Help] [answered]
#### Name:
Logan Webb
#### Group:
Kylo Ren's Rice Crispie Treats
#### Have you searched the other issue tickets? If not, do that first. 
- [x ] Yes
- If No, stop and search.

####  REQUIRED Add Tags to Issue Ticket and Assign to Project Board
 

- [x ] Added label to specific project/assignment that you need help with
- [x ] Added to `c46 Help-Thread` project board

#### REQUIRED Describe your issue:
>>I know with session storage the information is stores in the browser, in that specific tab. Im wondering if when a new customer registers, does that registration info need to go into the database permanently? Is it just the fact they they are currently logged in the only thing that is stored in session storage?

#### REQUIRED What have you googled? 

- [ ] 1. https://javascript.info/localstorage#sessionstorage
- [ ] 2. https://gist.github.com/bshaffer/4997874
- [ ] 3. https://stackoverflow.com/questions/52697055/how-to-use-sessionstorage-for-a-webpage-for-a-login-user
- [ ] 4.https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage
- [ ] 5.https://js.plainenglish.io/localstorage-vs-sessionstorage-c22164aff067

#### REQUIRED: What have you tried? Tell us everything you have tried. 
>> I served my branch and tried to register a new user to see if it threw any errors, it didnt.

#### REQUIRED: Copy/paste link TO THE FILE on your branch that you are having issues with

**https://github.com/nss-day-cohort-46/sprinkles-of-joy-kylo-ren-s-rice-krispie-treats/blob/f1b3f098cf5263ddb8b8c94cd7c3e112e0d76355/scripts/customers/LoginForm.js#L33**

#### Code Snippet

```
https://github.com/nss-day-cohort-46/sprinkles-of-joy-kylo-ren-s-rice-krispie-treats/blob/f1b3f098cf5263ddb8b8c94cd7c3e112e0d76355/scripts/customers/LoginForm.js#L33
```

**Comment by Scott47** [comment, marked as answer]
re: `I know with session storage the information is stores in the browser, in that specific tab. Im wondering if when a new customer registers, does that registration info need to go into the database permanently? Is it just the fact they they are currently logged in the only thing that is stored in session storage?`

@Loganwebb1994 When you register a customer, there should be an object with properties like id, name, email, etc. Session storage also stores a key value pair in the browser. I am going to migrate this issue to discussion. @SubtleCo & @kjk1325 are digging into this today.

**Comment by SubtleCo** [comment]
@kjk1325 and I found an interesting phenomenon where in at least one instance, Session Storage didn't return the customerId (using custom method authHelper.getCurrentUserId) instantaneously. Adding it to a .then() chain, however, allowed the variable to fill by the time we needed it. Is it possible that accessing session storage is either slightly sluggish, or somehow asynchronous?

**Reply by SubtleCo** [reply]
I lied! This is not asynchronous - We were accidentally defining customerId via an import statement on page load, so if a user was logged out, it would return NaN. Thanks @jisie for pointing that out!

**Reply by Scott47** [reply]
Scope really matters here. It's not asynchronous, but getting the item out of session storage in the moment you need it was tricky for @nch66862 , @kylieanyce and I. They had the same issue. The solution was a little different, I think. Anyone have before and after code snippets? Don't worry about it, if not. Just an interesting issue.
